Updates to readme, .env, and dependencies

AlexanderGriffin opened the pull request:
Related to task [#33943054](https://savaslabs.teamwork.com/#/tasks/33943054)

## Summary of changes

1. Added 'dotenv@v16.0.1' dependency to package.json and lockfile 
2. Added .env.sample and .env files
3. Changed require code for Pinata; there were some reference errors
4. Updated README.md file outlining details about installation, development, and configuration (Specifically for CORS)

## To test

- [x] Follow the README.md instructions to install/setup the proxy
- [x] Navigate to http://localhost:8080/ (or the location proxy was installed up at)
- [x] See 'Hello developers!' returned in the browser from the /GET request
- [x] Optionally: If testing in conjunction with a locally (or alternately) hosted Minterpress WP plugin, configure the .env file per the README.md instructions to allow requests from the plugin
- [x] Optionally: If the previous step was followed, use the 'Minterpress' plugin to connect to a wallet (provided it has testnet funds in the configured testnet) and mint and NFT

### Pull request reviewer

As the reviewer, I have verified the following:

- [x] I have tested the PR in the appropriate local / staging environment(s)

maddybvh reviewed (changes requested):
Thanks so much for all of the documentation and configuration updates, @AlexanderGriffin  major improvements!!

Everything worked perfectly for me following your instructions locally on `testnet` and `mainnet` with local wp and local proxy. However, that was already the case for me so I went the extra step to test with a hosted proxy (I deployed a new instance to heroku based on this branch) and hosted wordpress (on Pantheon). 

The error I ran into with the hosted version was: `ERR_INVALID_PROTOCOL` and I was able to track that down to the `http.get` call that creates a stream of the asset to be minted. 

I recreated and fixed this locally with the following steps (starting from this branch and `minterpress/feature/33943054-CORS-troubleshooting`.

- [ ] In `minterpress/assets/admin/js/helpers/mintHelpers.js` line 49, set the `remoteFileUrl` to [any https file location](https://dev-minterpress-wp.pantheonsite.io/wp-content/uploads/2021/12/Labs-Peering-Ava.png).
- [ ] Attempt to mint (you'll have to upload an image but will be overridden by the remote url.
- [ ] This should fail with `ERR_INVALID_PROTOCOL`.
- [ ] Go over to your proxy and change the get handler to `https.get(`
- [ ] Try again, minting should succeed!

To handle both http or https images, I suggest the following change in `index.ts`, which I've tested both locally and in hosted environments with success!

```
const http = require("http");
const https = require("https");
....
let httpHandler: any; 
httpHandler = url.match("^https://") ? https : http;
httpHandler.get(url, function (response: Response) {
```

Additionally, to get this working on Heroku, I needed to add a multer destination. I think this primes Heroku to allow temporary file uploads to the uploads/ directory.
```
const upload = multer({ dest: "uploads/" });
```

AlexanderGriffin commented on .env.sample:
```diff
@@ -0,0 +1,2 @@
+env=local
```
I added it for consistency with the plugin and because having an .env file without a env variable inside of it seemed odd. That being said, I am absolutely fine with removing it since it is not used directly but more there for extensibility and consistency.

AlexanderGriffin commented on .env:
```diff
@@ -0,0 +1 @@
+env=local
```
You are correct!

AlexanderGriffin commented on README.md:
```diff
@@ -70,7 +76,44 @@ starts the server on http://localhost:8080 in build mode without hot reloading.
 The application includes a Procfile and is ready to be deployed to Heroku. To deploy to Heroku, follow the instructions at [Creating Apps from the CLI](https://devcenter.heroku.com/articles/creating-apps).
 
-Prior to deploying, update the origin in the corsOptions to only allow requests from your client app.
+***IMPORTANT: Prior to deploying, the 'pluginBaseUrl' variable in the .env file most be updated with the URL plugin will be hosted / making requests from. By default, it will be set to a wildcard * allowing access from anywhere***
```
Removed this note for now.

AlexanderGriffin commented on src/index.ts:
```diff
@@ -31,6 +36,7 @@ app.get("/", (req: Request, res: Response) => {
 app.post("/pin", upload.none(), async (req: Request, res: Response) => {
   console.log("Testing Pinata auth.");
   try {
+    console.log(process.env.pluginBaseUrl)
```
Removed.

AlexanderGriffin commented:
Hi Maddy,

I recreated and fixed this locally with the following steps (starting from this branch and `minterpress/feature/33943054-CORS-troubleshooting`.

- [x] In `minterpress/assets/admin/js/helpers/mintHelpers.js` line 49, set the `remoteFileUrl` to [any https file location](https://dev-minterpress-wp.pantheonsite.io/wp-content/uploads/2021/12/Labs-Peering-Ava.png).
- [x] Attempt to mint (you'll have to upload an image but will be overridden by the remote url.
- [x] This failed with `ERR_INVALID_PROTOCOL`.
- [x] Went to proxy and changed the get handler to `https.get(`
- [x] Minting succeeded
- [x] Changed the handler to suggested code to be dynamic
- [x] Minting succeeded
- [x] Replaced `minterpress/assets/admin/js/helpers/mintHelpers.js` line 49 with original value
- [x] Minting succeeded.

These changes have been committed and I re-requested your review.

maddybvh reviewed (approved):
Looks great! Thanks for making those changes!
